pkg/hubble/relay/observer/server.go

```diff
@@ -318,6 +302,13 @@ func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusR
 					"error": err,
 					"peer":  p,
 				}).Warning("Failed to retrieve server status")
+				mu.Lock()
+				numConnectedNodes--
+				numUnavailableNodes++
+				if len(unavailableNodes) < numUnavailableNodesReportMax {
+					unavailableNodes = append(unavailableNodes, p.Name)
+				}
+				mu.Unlock()
```

Comment:
if `mu` is protecting `numConnectedNodes`, `numUnavailableNodes`, and `unavailableNodes`, then we should lock it when accessing them from the "main" goroutine in the `for` loop above right? (lines 286, 290-292, and 295).